stdlib/LinearAlgebra/src/blas.jl

```diff
@@ -130,6 +128,24 @@ function set_num_threads(n::Integer)
     return nothing
 end
 
+"""
+    get_num_threads()
+
+Get the number of threads the BLAS library is using.
+"""
+function get_num_threads()
+    blas = LinearAlgebra.BLAS.vendor()
+    if blas === :openblas || blas === :openblas64
+        return ccall((@blasfunc(openblas_get_num_threads), libblas), Cint, ())
+    elseif blas == :mkl
+        return ccall((:mkl_get_max_threads, libblas), Cint, ())
+    elseif Sys.isapple()
+        return Base.parse(Cint, ENV["VECLIB_MAXIMUM_THREADS"])
```

Comment:
This doesn't run for me, on a mac.

Reply:
try this
```julia
@static if Sys.isapple()
        ntreads = Base.tryparse(Cint, get(ENV, "VECLIB_MAXIMUM_THREADS", "$(Sys.CPU_THREADS)"))
        nthreads isa Nothing && return Sys.CPU_THREADS
        return nthreads
    end
```

Reply:
what do you mean?

Reply:
i was referring to @mcabbott to check if the updated code above runs on his mac

Reply:
Oh sorry. It is still magical to me, that this block needs to be elided on mac. Can somebody explain what exactly fails on mac?

Reply:
Why is this this block needed at all?

As far as I know, `@static if` is only useful to shield things like macros which aren't defined.

Reply:
So what does it shield in the case at hand?

Reply:
That option is what's used in existing `set_num_threads` to set number of threads for MacOS

Reply:
> So what does it shield in the case at hand?

@jw3126. I guess you could skip the `@static` and write it this way instead.(since all constructs are valid on all OS platforms)
```julia
if Sys.isapple()
        ntreads = Base.tryparse(Cint, get(ENV, "VECLIB_MAXIMUM_THREADS", "$(Sys.CPU_THREADS)"))
        nthreads isa Nothing && return Sys.CPU_THREADS
        return nthreads
end
```

Reply:
Still seems a bit baroque to parse "Sys.CPU_THREADS" as a string. Shouldn't it be more like 
```
elseif Sys.isapple()
    val = get(ENV, "VECLIB_MAXIMUM_THREADS", nothing)
    isnothing(val) || return parse(Cint, val)
end

@warn "unable to read number of BLAS threads" maxlog=1
return Sys.CPU_THREADS
```